fits2caom2/src/main/java/ca/nrc/cadc/caom2/fits/wcs/Wcs.java

```diff
@@ -473,31 +493,28 @@ public static CoordFunction2D getCoordFunction2D(String utype, FitsMapping mappi
             pc21 = 0.0;
             pc22 = 1.0;
 
-            String crota1_kw = mapping.getKeywordValue("CROTA" + posAxis1);
-            String crota2_kw = mapping.getKeywordValue("CROTA" + posAxis2);
-            if (crota1_kw != null && crota2_kw == null) {
-                crota2_kw = crota1_kw;
-            } else if (crota1_kw == null && crota2_kw != null) {
-                crota1_kw = crota2_kw;
-            }
-            if (crota1_kw == null && crota2_kw == null) {
-                String crota_kw = mapping.getKeywordValue("CROTA");
-                if (crota_kw != null) {
-                    crota1_kw = crota_kw;
-                    crota2_kw = crota_kw;
+            String crota_kw = mapping.getKeywordValue("CROTA" + posAxis1);
+            Double crota = parseDouble(crota_kw, "CROTA" + posAxis1);
+            if (crota == null) {
+                crota_kw = mapping.getKeywordValue("CROTA" + posAxis2);
+                crota = parseDouble(crota_kw, "CROTA" + posAxis2);
+                if (crota != null) {
+                    crota = 90 - crota;
                 }
             }
-            Double crota1 = parseDouble(crota1_kw, "CROTA" + posAxis1);;
-            Double crota2 = parseDouble(crota2_kw, "CROTA" + posAxis1);;
+            if (crota == null) {
+                crota_kw = mapping.getKeywordValue("CROTA");
+                crota = parseDouble(crota_kw, "CROTA");
+            }
 
             String crotsign_kw = mapping.getKeywordValue("CROTSIGN");
             double crotsign = (crotsign_kw == null ? 1.0 : parseDouble(crotsign_kw,"CROTSIGN"));
 
-            if (crota1 != null && crota2 != null) {
-                pc11 = pc11 * Math.cos(crota1) * crotsign;
-                pc12 = pc12 * Math.sin(crota1) * crotsign;
-                pc21 = pc21 * Math.sin(crota2) * crotsign;
-                pc22 = pc22 * Math.cos(crota2) * crotsign;
+            if (crota != null) {
+                pc11 = Math.cos(crota) * crotsign;
+                pc12 = -Math.sin(crota) * crotsign;
```

Comment:
I would really prefer
```pc12 = -1.0 * Math.sin(crota) * crotsign;```
because that - sign by itself could easily be missed/removed. In fact, I'm surprised that checkstyle allows it, but maybe checkstyle is not enabled for fits2caom2 (which is fine).

Reply:
Updated, and check style is not enabled for fits2caom because it has not been converted to the new code style.